scriptworker/gpg.py

```diff
@@ -1322,8 +1313,48 @@ def verify_signed_git_commit_output(output):
     verify_signed_git_commit_output(output)
 
 
+# last_good_git_revision_file functions {{{1
+def get_last_good_git_revision(context):
+    """This returns the contents of the config['last_good_git_revision_file'], if it exists.
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context.
+
+    Returns:
+        str: the latest good git revision, if the file exists
+        None: if the file doesn't exist
+    """
+    result = None
+    if os.path.exists(context.config['last_good_git_revision_file']):
+        with open(context.config['last_good_git_revision_file'], "r") as fh:
+            result = fh.read().rstrip()
+    return result
+
+
+def write_last_good_git_revision(context, revision):
+    """This writes `revision` to config['last_good_git_revision_file'].
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context.
+        revision (str): the last good git revision
+
+    Raises:
+        ScriptWorkerGPGException: on failure
+    """
+    log.info(
+        "writing last_good_git_revision {} to {}...".format(
+            revision, context.config['last_good_git_revision_file']
+        )
+    )
+    try:
+        with open(context.config['last_good_git_revision_file'], "w") as fh:
+            print(revision, file=fh, end='')
+    except OSError as exc:
+        raise ScriptWorkerGPGException(str(exc))
```

Comment:
Now that we're only calling this from external endpoints, I think we don't have to catch the `OSError` at all.  It does the trick and kills the process, while removing the lockfile in the `finally` block.  I removed the try/except block.